src/main/kotlin/com/hibob/academy/resource/petsResource.kt

```diff
@@ -0,0 +1,49 @@
+package com.hibob.academy.resource
+
+import com.hibob.academy.types.Pet
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.RequestBody
+import java.util.*
+
+@Controller
+@Path("/api/noy/pets")
+@Produces(MediaType.APPLICATION_JSON)
+class PetsResource {
+
+// create
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/newPet")
+    fun addPet(@RequestBody pet: Pet): Response {
+        Response.status(Response.Status.CREATED).build()
+        return Response.ok(pet).build()
+    }
+
+// update
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/updatePetName")
+    fun updateName(@RequestBody name: String): Response {
+        Response.status(Response.Status.ACCEPTED).build()
+        return Response.ok(name).build()
+    }
+
+// retrieve
+    @GET
+    @Path("/{petId}/type")
+    fun getPetType(@PathParam("petId") petId: UUID): Response {
```

Comment:
Just for the sake of learning as much as possible, try to fake (mock) that the pet you're trying to find doesn't exist. How would you handle it?